packages/middleware/src/apiClientFactory/applyContextToApi.ts

```diff
@@ -35,32 +35,37 @@ const applyContextToApi = <
     (prev, [callName, fn]) => ({
       ...prev,
       [callName]: async (...args: Parameters<typeof fn>) => {
-        const extendQuery = createExtendQuery(context);
-        const transformedArgs = await hooks.before({ callName, args });
-        const apiClientContext = { ...context, extendQuery };
-        if (isExtensionEndpointHandler(fn)) {
-          if (context.res.locals?.alokai?.metadata?.scope) {
-            context.res.locals.alokai.metadata.scope.extensionName =
-              fn._extensionName;
-          } else {
-            const logger = getLogger(context.res);
-            logger.warning(
-              `Alokai's metadata object is missing in the context under 'res.locals.alokai'. 
-              This could indicate that the extension's scope or metadata has not been properly initialized. 
-              Without this metadata, certain custom API client functionalities may not work as expected, 
-              including tracking of extension-specific actions or data. 
-              Please ensure that the Alokai metadata object is correctly set up in 'res.locals.alokai' before proceeding with the request.
-              
-              Steps to troubleshoot:
-              1. Verify if content of res.locals hasn't been overwritten, instead of extended.
-              2. If you're unsure, please consult Alokai's team for further assistance or review the source code implementation.
-
-              Call Name: ${callName}
-              Function Name: ${fn.name || "Unnamed function"}`
-            );
-          }
-        }
-        const response = await fn(apiClientContext, ...transformedArgs);
+        const transformedArgs = await hooks.before({
+          callName,
+          args,
+        });
+        const apiClientContext = {
+          ...context,
+          extendQuery: createExtendQuery(context),
+        };
+        const response = await fn(
+          {
+            ...apiClientContext,
+            logger: injectMetadata(getLogger(context), (metadata) => {
```

Comment:
Maybe we can build this logger somewhere else? Like in  separate function? It will improve the readability.